contrib/depends/Makefile

```diff
@@ -111,7 +116,7 @@ $(host_arch)_$(host_os)_native_toolchain?=$($(host_os)_native_toolchain)
 include funcs.mk
 
 toolchain_path=$($($(host_arch)_$(host_os)_native_toolchain)_prefixbin)
-final_build_id_long+=$(shell $(build_SHA256SUM) toolchain.cmake.in)
+final_build_id_long+=:[sha256sum]:$(shell $(build_SHA256SUM) toolchain.cmake.in)
```

Comment:
Is there a particular convention you want these templates to follow? Seems it's a bit all over the map, as I see `:[key]:value` here, `:[key]value:` in contrib/depends/funcs.mk, etc. I suggest picking a convention which both a human and machine can read easily (i.e. a simple procedure, which can easily be implemented in code and by the human brain, not necessarily the fastest). With that in mind, if you want to use `:` as a separator between pairs, it probably shouldn't also be used after the key.

Reply:
The colon separator is replaced with a newline in the final output (`depends-hashes.txt`) and is easily diff-able:

```
[boost_all_file_checksums]
5419a1f3ae6346deeb4ad5d60964f09a2afd443e75f86bff52032cfd6b1e5189 
2948dc359030cbd8fe411bc544212d7cf7c3308643aec3b550ddda17d512802f 
884976d97b7a386c4c0c4aeb3e0ffe40848481d0fd5f379044815ca1e24f5622 
0ddb4c9faeb4443251af83ef8924cc2579c225d8e3cc7a1b65fc206e22b0b4f2 
20a3fff6d950af9b70daf3ab005f6ba19c0b069970470e1ca81ee1e626ffe680 
897adbccc9b4bd22585dde1c443a119eaa5e01641b43b3d5960f7137197b3b54 
7284e8b6e1471f20c0e12d3f99f4301b1bc4368fd5b7d330557386939a136d24

[recipe]
unbound-1.19.1-8a76263f0ba3f0bc927efe0b29ff3067a58dbcf4d45126b42cef26bffde55c26-Release
[deps]
expat-2.6.0-ea84dd3e3694465ec4a470c96e79089ed11cbc13f43a3201094dda24c0c11814 
openssl-3.0.13-348c61a404a4575b3270d742dce1e58b5c6b0e140d67030fe20b5c15d5e6cd38
[x86_64_linux_id]
/gnu/store/03rfvpz53ab1klw20j5padpxmygcmksl-profile

[sha256sum]
1e82b195b37594b1753317b0b864dd036af3e1b7812dc6825df7362872158933  toolchain.cmake.in
```

I can probably turn this into valid yaml with minor modifications to the code, so that it's easily machine readable.

Reply:
No need to make it valid YAML, the format you have is easily parseable, I misread one of the foreach statements (didn't notice it was adding a newline with echo).